guides/doc-Managing_Hosts/topics/Global_Registration.adoc

```diff
@@ -0,0 +1,138 @@
+Main power of this feature came from utilization of provisioning templates,
+which gives user complete control over the process of the host registration.
+
+
+.Prerequisites
+- Foreman user must have `create_hosts` permission
+- Host OS must have assigned Registration template.
+  Go to the template detail > Associations & select the OS
+  Then on OS detail (Hosts > Operating Systems) set default registration template
+- Root privileges on the host that's going to be registered.
+
+
+.Procedure
+. Login to {ProjectName}
+. Go to Hosts > All Hosts > Register Host
+. Set required parameters and generate the command
+. Log in to the Host
+. Run command generated in {ProjectServer}
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# Example:
+curl http://_{foreman-example-com}_/register | bash
+----
++
+
+
+.Templates & snippets
+
+[cols=2*]
+|===
+|Global Registration Template
+|Contains steps for registration of host to the Foreman. Rendered when accessing `/register` endpoint.
+
+|Host Registration Template
+|In Foreman named `Linux registration default`, contains commands for Host setup.
+
+|`remote_execution_ssh_keys` snippet
+| Used in Host registration template, deploy SSH keys to the host. Only when `host_registration_remote_execution` parameter is `true`. Part of Host registration template
+
+|`insights` snippet
+| Download & install insights client. Only for Red Hat OS & when global parameter `host_registration_insights` is set to true. Part of Host registration template
+|===
+
+
+.Registration parameters
+* Organization
+* Location
+* Host Group
+* Operating System - No need to specify for registration with subscription-manager
+* Proxy
+* Setup Insights - Install insights client & register host. Override value of `host_registration_insights` global parameter on the Host level. Plus only for Red Hat OS.
+* Remote Execution - Override value of `host_registration_remote_execution` global parameter on the Host level.
+* Token lifetime
+
+
+.Settings
+* For changing default Global Registration template go to Administer > Settings > Provisioning
+
+
+.Global Parameters
+* `host_registration_remote_execution` - used in `remote_execution_ssh_keys` snippet, default value is `true`.
+* `host_registration_insights` - used in `insights` snippet, default value is `false` for the Foreman and `true` for the Satellite
+
+
+.With Smart proxy
+For registration through proxy you need to enable Registration module
+[options="nowrap", subs="+quotes,attributes"]
+----
+foreman-installer --foreman-proxy-registration
+----
+
+Then you can call `/register` endpoint on the proxy:
+[options="nowrap", subs="+quotes,attributes"]
+----
+curl http://proxy.example.com/register | bash
+----
+
+.SSL
+All calls in the templates to the Foreman are secure & use https.
+However, the first call to the Foreman (`curl /register`) requires to have certificate authority
+installed on the host or you can call the register command with `--insecure` parameter.
```

Comment:
it would be great to mention how to fetch that SSL cert from the client machine or how to get it on Foreman host (what path to take it from). It's good you mention the `--insecure`, it would also benefit from saying that GRT deploys the certificate for the registration so the rest of the communication is then properly secured.